rust/src/lib/ovsdb/global_conf.rs

```diff
@@ -33,46 +36,193 @@ fn convert_map(
-            }
+impl OvsDbConnection {
+    pub(crate) fn get_global_conf(
+        &mut self,
+    ) -> Result<OvsDbGlobalConfig, NmstateError> {
+        let reply = self.transact(&OvsDbMethodTransact {
+            db_name: OVS_DB_NAME.to_string(),
+            operations: vec![OvsDbOperation::Select(OvsDbSelect {
+                table: GLOBAL_CONFIG_TABLE.to_string(),
+                conditions: vec![],
+                columns: Some(vec!["external_ids", "other_config"]),
+            })],
+        })?;
+
+        if let Some(global_conf) = reply
+            .as_array()
+            .and_then(|reply| reply.first())
+            .and_then(|v| v.as_object())
+            .and_then(|v| v.get("rows"))
+            .and_then(|v| v.as_array())
+            .and_then(|v| v.first())
+            .and_then(|v| v.as_object())
+        {
+            Ok(global_conf.into())
+        } else {
+            let e = NmstateError::new(
+                ErrorKind::PluginFailure,
+                format!(
+                    "Invalid reply from OVSDB for querying \
+                    {GLOBAL_CONFIG_TABLE} table: {reply:?}"
+                ),
+            );
+            log::error!("{}", e);
+            Err(e)
         }
-        row.insert(
-            "external_ids".to_string(),
-            Value::Array(vec![
-                Value::String("map".to_string()),
-                Value::Array(value_array),
+    }
+}
+
+fn append_purge_action(operations: &mut Vec<OvsDbOperation>, column: &str) {
+    let mut row = HashMap::new();
+    row.insert(
+        column.to_string(),
+        Value::Array(vec![
+            Value::String("map".to_string()),
+            Value::Array(Vec::new()),
+        ]),
+    );
+    operations.push(OvsDbOperation::Update(OvsDbUpdate {
+        table: GLOBAL_CONFIG_TABLE.to_string(),
+        conditions: Vec::new(),
+        row,
+    }))
+}
+
+// Use `delete` mutator for Value set to None, otherwise use `insert`.
+fn append_mutations(
+    mutations: &mut Vec<OvsDbMutation>,
+    column: &str,
+    data: &HashMap<String, Option<String>>,
+) {
+    // All key should be deleted first because `insert` does not override
+    // existing values.
+    let delete_keys: Vec<Value> =
+        data.keys().map(|k| Value::String(k.to_string())).collect();
+    if !delete_keys.is_empty() {
+        mutations.push(OvsDbMutation {
+            column: column.to_string(),
+            mutator: "delete".to_string(),
+            value: Value::Array(vec![
+                Value::String("set".to_string()),
+                Value::Array(delete_keys),
             ]),
-        );
-        let mut value_array = Vec::new();
-        for (k, v) in ovs_conf.other_config.iter() {
-            if let Some(v) = v {
-                value_array.push(Value::Array(vec![
+        })
+    }
+    let insert_values: Vec<Value> = data
+        .iter()
+        .filter_map(|(k, v)| {
+            v.as_ref().map(|v| {
+                Value::Array(vec![
```

Comment:
How about also logging the keys being inserted?

Reply:
Added `log::debug!("Inserting key {} value {} into {}", k, v, column);`